openedx/core/lib/api/permissions.py

```diff
@@ -80,3 +86,64 @@ def has_object_permission(self, request, view, obj):
         return (request.user.is_staff or
                 CourseStaffRole(obj.course_id).has_user(request.user) or
                 request.method in permissions.SAFE_METHODS)
+
+
+def get_profile_visibility(user_profile, user, configuration=None):
```

Comment:
How about you leave the methods in user_api/accounts/serializers.py and put a wrapper for visible_fields in user_api/accounts/api.py (simply calls the implementation in serializers) for other djangoapps to use?